### PR TITLE
Refactor db migration

### PR DESCRIPTION
**Pull request by thomash-acinq**
Simplify the migration by not duplicating code. Should be less error-prone when adding new migration.

### CODE REVIEW

**Review by t-bast — DISMISSED** [review]

**Comment by pm47** [comment]
Another thing that would make db updates a little less of a hassle would be to move `val CURRENT_VERSION = 8` to the companion object. This would save updating test files at multiple place.

I may have another idea for the migration process, will let you know.

**Review by pm47 — APPROVED** [review]

**Comment by thomash-acinq** [comment]
I've moved `CURRENT_VERSION` to the companion object.

File: eclair-core/src/test/scala/fr/acinq/eclair/db/AuditDbSpec.scala [hunk 1]
```diff
@@ -229,7 +229,7 @@ class AuditDbSpec extends AnyFunSuite {
         }
       },
       dbName = "audit",
-      targetVersion = 6,
+      targetVersion = SqliteAuditDb.CURRENT_VERSION,
```
Comment (pm47) [inline eclair-core/src/test/scala/fr/acinq/eclair/db/AuditDbSpec.scala:232]: The line above could also be updated to `SqliteAuditDb.DB_NAME`.

**Review by pm47 — COMMENTED** [review]
The concern I have now is that nothing prevents a database file to mistakenly import the wrong db object.

**Comment by thomash-acinq** [comment]
I don't get your concern about importing the wrong db object, can you give an example?

**Comment by pm47** [comment]
> I don't get your concern about importing the wrong db object, can you give an example?

If we mistakenly import e.g. `import PgAuditDb._` from `PgChannels`, then the migration will be completely broken. Tests should catch that though.

**Review by pm47 — APPROVED** [review]

**Review by t-bast — APPROVED** [review]
:+1:

**Pull request merged**